keras/utils/generic_utils.py

```diff
@@ -313,8 +313,11 @@ def update(self, current, values=None, force=False):
                         bar += '='
                 bar += ('.' * (self.width - prog_width))
                 bar += ']'
-                sys.stdout.write(bar)
-                self.total_width = len(bar)
+            else:
+                bar = '%7d/Unknown' % current
```

Comment:
Why not just `%d`?

Reply:
Just following the other branch of the conditional. Also, even with \r, %7 will look better if piped somewhere.

Reply:
Looking at the other branch of the conditional, I noticed that computing %width for self.target is redundant. I added a fix for it to this PR (which will also trigger Travis restart, which fails due to unrelated reason).